src/utils/wallet.js

```diff
@@ -438,53 +647,266 @@ class Wallet {
+        setRequest({})
+        return await this.postSignedJson('/2fa/init', {
             accountId,
             method
         });
     }
 
-    async validateSecurityCode(accountId, method, securityCode) {
-        return await this.postSignedJson('/account/validateSecurityCode', {
+    async reInitTwoFactor(accountId, method) {
+        // clear any previous requests in localStorage (for verifyTwoFactor)
+        setRequest({})
+        return this.sendTwoFactor(accountId, method)
+    }
+
+    async sendTwoFactor(accountId, method, requestId = -1, data = {}) {
+        if (!accountId) accountId = this.accountId
+        if (!method) method = await this.get2faMethod()
+        // add request to local storage
+        setRequest({ accountId, requestId, data })
+
+        console.log(accountId, requestId, data)
+        console.log('tempTwoFactorAccount', tempTwoFactorAccount)
+
+        const serverResponse = this.postSignedJson('/2fa/send', {
             accountId,
             method,
-            securityCode
+            requestId,
+            data
+        }).catch((e) => console.log('/2fa/send failed', e));
+        
+        if (requestId !== -1) {
+            // we know the requestId of what we want to confirm so pop the modal now and wait on that
+            return new Promise((resolve) => {
+                store.dispatch(promptTwoFactor((verified) => {
+                    if (verified) {
+                        tempTwoFactorAccount = null
+                    }
+                    resolve(verified)
+                }))
+            })
+        } else {
+            // wait for sever response to reture
+            return await serverResponse
+        }
+    }
+
+    async resendTwoFactor(accountId, method) {
+        if (!accountId) accountId = this.accountId
+        if (!method) method = await this.get2faMethod()
+        const requestData = getRequest()
+        let { requestId, data } = requestData
+        if (!requestId && requestId !== 0) {
+            console.log('no pending multisig requestId found, assuming account setup')
+            requestId = -1
+        }
+        return this.sendTwoFactor(accountId, method, requestId, data)
+    }
+
+    // requestId is optional, if included the server will try to confirm requestId
+    async verifyTwoFactor(accountId, securityCode) {
+        const requestData = getRequest()
+        console.log(requestData)
+        let { requestId } = requestData
+        if (!requestId && requestId !== 0) {
+            console.log('no pending multisig requestId found, assuming account setup')
+            requestId = -1
+        }
+        // try to get a accountId for the request
+        if (!accountId) accountId = requestData.accountId || this.accountId
+        if (!accountId) {
+            console.error('no pending multisig accountId found')
+            return
+        }
+        return await this.postSignedJson('/2fa/verify', {
```

Comment:
I don't understand why all above logic is needed. Shouldn't we only try to verify valid requests in the first place?

Reply:
I believe this has been updated in twoFactor.js. The logic was to reduce endpoints in contract helper, overloading send and verify to also handle the creation of a 2fa method.